CONTRIBUTING.md

```diff
@@ -1,65 +1,119 @@
 
-To set up an initial development environment:
+To set up a local copy of the repository for development:
 
-    git clone https://github.com/canonical/charmcraft.git
-    cd charmcraft
-    uv sync --all-extras
+```bash
+git clone https://github.com/canonical/charmcraft.git
+cd charmcraft
+uv sync --all-extras
+```
 
 If you need `python-apt`, add `https://people.canonical.com/~lengau/pypi/` as an
 extra index URL.
 
 You will need a copy of `ruff` installed. On many Linux distributions, you
 can install ruff with:
 
-    sudo snap install ruff
+```bash
+sudo snap install ruff
+```
 
 Otherwise, you can install ruff in your virtual environment with:
 
-    uv tool install ruff
+```bash
+uv tool install ruff
+```
 
 
-## Developing against Charmcraft source
+## Develop for the Charmcraft source
 
 Make changes as appropriate. Some existing ideas are in the
 [Github Issues](https://github.com/canonical/charmcraft/issues)
 
 To test locally:
 
-    CHARMCRAFT_DEVELOPER=1 python -m charmcraft
+```bash
+CHARMCRAFT_DEVELOPER=1 python -m charmcraft
+```
 
 When you're done, make sure you run the tests.
 
-You can do so with
+You can do so with:
 
-    uv sync
-    uv run pytest
+```bash
+uv sync
+uv run pytest
+```
 
 Contributions welcome!
 
-## Contributing to documentation
 
-Charmcraft uses Sphinx for its documentation, so contributing to the documentation
-is similar to contributing to the code. To begin, fork and clone the repository.
-Once in the repository directory, run:
+## Contribute to the documentation
 
-    make setup
+Charmcraft stores its documentation source in the repository and tests it for errors.
+Contributing to the documentation is similar to contributing to the code.
 
-to bring in any build dependencies and create a virtual environment. Before proceeding,
-check that the documentation builds and lints correctly on your machine:
 
-    make lint-docs
-    make docs
+### Set up the documentation environment
 
-If both of those commands succeed, you can make a branch and start writing docs.
-To continuously build and locally host the documentation while writing docs, run:
+First, [download the repository and tools](#set-up-the-development-environment).
 
-    make docs-auto
+Then, from the root of the repository, install the dependencies and build environment:
 
-This will run `sphinx-autobuild` and provide you with a link to view the updated
-documentation in your browser.
+```bash
+make setup
+```
+
+
+### Write the docs
+
+The Charmcraft documentation follows [Diátaxis](https://diataxis.fr), and adheres to the
+conventions of the [Canonical Documentation Style
+Guide](https://docs.ubuntu.com/styleguide/en).
+
+The documentation source is written in reStructuredText, and is formatted according to
+[Canonical's reStructuredText
+style](https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide). It uses Sphinx for linting and building.
+
```

Comment:
Here I'd add:

> Except for the tutorial, which should give a complete end-to-end story, including any necessary bits such as Juju, Rockcraft, or Ops, the contents of the docs should be confined to functionality directly provided by Charmcraft, with "See first" and "See next" type links to the related content in Juju, Rockcraft, or Ops docs wherever relevant.

Reply:
I agree that we need clearer guidelines.

For now, though, I agree that this would be too much fine detail for a contributing.md. Along your thinking, though, we might be able to give some hints about the interconnection:

> ... It uses Sphinx for linting and building, and employs the [intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html) extension to link to the related Juju and ops library documentation.

This would at least plant the seed for readers that there's a more complex setup going on.

Reply:
[59b8067](https://github.com/canonical/charmcraft/pull/2101/commits/59b80675d576ade4878270eca76fcd0b03478430)